### PR TITLE
A4A: Implement the 'Add site via Jetpack connection' modal.

### PR DESCRIPTION
**Pull request by jkguidaven**
This pull request implements the modal for adding a site via Jetpack connection, which will be used in the central hub site selector.

Screenshot 2024-07-02 at 1 57 29 AM

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/679

## Proposed Changes

* Implement the `A4AThemedModal` component as a reusable skeleton for a themed-based modal.
* Implement the `AddSiteViaJetpackConnectionModal` component based on the new reusable themed-based modal.

## Testing Instructions

* Use the A4A live link and go to `/overview` page.
* Click the **Add site** dropdown menu at the top right.
* Click the **Via Jetpack** option to show the modal.
* Test that the modal functions as expected.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/components/a4a-themed-modal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import clsx from 'clsx';
+import { ReactNode } from 'react';
+
+import './style.scss';
+
+type Props = {
+	className?: string;
+	children: ReactNode;
+	onClose?: () => void;
+	sidebarImage?: string;
```
Comment (yashwin) [inline client/a8c-for-agencies/components/a4a-themed-modal/index.tsx]: Can we rename this to something else? For me, this feels slightly confusing.

How about `leftPanelImage` or `leftImage` or simply `modalImage`?

**Review by yashwin — APPROVED** [review]
LGTM. 

I left a comment, but it is not a blocker.

File: client/a8c-for-agencies/components/a4a-themed-modal/index.tsx (hunk 1 above)
Comment (jkguidaven) [reply client/a8c-for-agencies/components/a4a-themed-modal/index.tsx]: `modalImage` sounds perfect. 👍

**Comment by jkguidaven** [comment]
https://agencies.automattic.com/overview

**Review by yashwin — APPROVED** [review]
Thanks for addressing my comment!

LGTM now 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/15854493

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @jkguidaven for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/15854493).

**Pull request merged**